### PR TITLE
Add flutter_source_test

### PR DESCRIPTION
**Pull request by digiter**
- Added a unit test for flutter source.
- Copied a fake datastore from Cocoon for testing.
- Also `dartfmt` everything, sorry for the extra changes.

### CODE REVIEW

File: test/flutter_source_test.dart [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      fakeDB = FakeDatastoreDB();
+      source = FlutterSource(fakeDB);
+    });
+
+    test('sets source time', () async {
+      MetricPoint point1 = MetricPoint(
+        0.0,
+        <String, dynamic>{'gitRevision': 'sha1'},
+        'flutter-center',
+        DateTime.fromMicrosecondsSinceEpoch(20),
```
Comment (liyuqian) [inline test/flutter_source_test.dart]: Shall we leave `sourceTime` as `null` in the constructor to test that `updateSourceTime` actually sets it?

File: test/src/fake_datastore.dart [hunk 2]
```diff
@@ -0,0 +1,306 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// This is copied from https://github.com/flutter/cocoon/blob/master/app_dart/test/src/datastore/fake_datastore.dart
```
Comment (liyuqian) [inline test/src/fake_datastore.dart:5]: This is good here. Although it seems to be better if we published that `fake_datastore` as a pub package so we can just use it by adding a dependency here. That way, we'll get the updated code if someone improved or fixed the `fake_datastore` code.

**Review by liyuqian — APPROVED** [review]
LGTM!

File: test/flutter_source_test.dart (hunk 1 above)
Comment (digiter) [reply test/flutter_source_test.dart]: Done, good idea.

File: test/src/fake_datastore.dart (hunk 2 above)
Comment (digiter) [reply test/src/fake_datastore.dart:5]: It's better to consider that when we productionize metrics_center. It's out of scope of this PR right now.

**Pull request merged**